hazelcast-client/src/main/java/com/hazelcast/client/impl/statistics/Statistics.java

```diff
@@ -0,0 +1,235 @@
+            if (!name.startsWith("/")) {
+                 pathPrefix += "/";
+            }
+            pathPrefix += name;
+            if (!name.endsWith("/")) {
+                pathPrefix += "/";
+            }
+            NearCacheStats nearCacheStats = nearCache.getNearCacheStats();
+            stats.put(pathPrefix + "CreationTime", Long.toString(nearCacheStats.getCreationTime()));
+            stats.put(pathPrefix + "Evictions", Long.toString(nearCacheStats.getEvictions()));
+            stats.put(pathPrefix + "Hits", Long.toString(nearCacheStats.getHits()));
+            stats.put(pathPrefix + "LastPersistenceDuration", Long.toString(nearCacheStats.getLastPersistenceDuration()));
+            stats.put(pathPrefix + "LastPersistenceKeyCount", Long.toString(nearCacheStats.getLastPersistenceKeyCount()));
+            stats.put(pathPrefix + "LastPersistenceTime", Long.toString(nearCacheStats.getLastPersistenceTime()));
+            stats.put(pathPrefix + "LastPersistenceWrittenBytes", Long.toString(nearCacheStats.getLastPersistenceWrittenBytes()));
+            stats.put(pathPrefix + "Misses", Long.toString(nearCacheStats.getMisses()));
+            stats.put(pathPrefix + "OwnedEntryCount", Long.toString(nearCacheStats.getOwnedEntryCount()));
+            stats.put(pathPrefix + "Expirations", Long.toString(nearCacheStats.getExpirations()));
+            stats.put(pathPrefix + "OwnedEntryMemoryCost", Long.toString(nearCacheStats.getOwnedEntryMemoryCost()));
+            stats.put(pathPrefix + "LastPersistenceFailure", nearCacheStats.getLastPersistenceFailure());
+        }
+    }
+
+    private void registerStaticStatistics() {
+        // The client may have already connected, hence, send the stat at least once at startup
+        sendStats(staticStats);
+
+        client.getLifecycleService().addLifecycleListener(new LifecycleListener() {
+            @Override
+            public void stateChanged(LifecycleEvent event) {
+                if (LifecycleEvent.LifecycleState.CLIENT_CONNECTED.equals(event.getState())) {
+                    sendStats(staticStats);
+                }
+            }
+        });
+    }
+
+    private void sendStats(Map<String, String> oneTimeStats) {
+        ClientMessage request = ClientStatisticsCodec.encodeRequest(oneTimeStats.entrySet());
+        Address ownerConnectionAddress = client.getClientClusterService().getOwnerConnectionAddress();
+        if (null != ownerConnectionAddress) {
+            try {
+                new ClientInvocation(client, request, ownerConnectionAddress).invoke().get();
+            } catch (Exception e) {
+                // suppress exception, do not print too many messages
+                logger.finest("Could not send stats " + e.getMessage());
+            }
+        }
+    }
+
+    private void registerPeriodicMetrics() {
+        RuntimeMetricSet.register(metricsRegistry);
+        OperatingSystemMetricSet.register(metricsRegistry);
```

Comment:
Having a single shared set of metrics and just filtering out the elements you need make more sense. Duplication a pain. 

Also I would not grab all probes; but just the focus on the most important ones.